internal/generator/backend/template/representations/r1cs.sparse.go.tmpl

```diff
@@ -245,38 +251,80 @@ func (cs *SparseR1CS) IsSolved(witness *witness.Witness, opts ...backend.ProverO
 	return err
 }
 
+// GetConstraints return a list of constraint formatted as in the paper
+// https://eprint.iacr.org/2019/953.pdf section 6 such that
+// qL⋅xa + qR⋅xb + qO⋅xc + qM⋅(xaxb) + qC == 0
+// each constraint is thus decomposed in [5]string with
+// 		[0] = qL⋅xa
+//		[1] = qR⋅xb
+//		[2] = qO⋅xc
+//		[3] = qM⋅(xaxb)
+//		[4] = qC
 func (cs *SparseR1CS) GetConstraints() [][]string {
 	var r [][]string 
```

Comment:
We know the number of returned strings ahead of time. We could allocate the whole slice to reduce the number of allocations in the loop.